src/Avalonia.FuncUI/DSL/Flyout.fs

```diff
@@ -1,31 +1,127 @@
+        static member showMode<'t when 't :> PopupFlyoutBase>(value: FlyoutShowMode) : IAttr<'t> =
             AttrBuilder<'t>.CreateProperty<FlyoutShowMode>(PopupFlyoutBase.ShowModeProperty, value, ValueNone)
 
+        static member overlayInputPassThroughElement<'t when 't :> PopupFlyoutBase>(value: IInputElement) : IAttr<'t> =
+            AttrBuilder<'t>.CreateProperty<IInputElement>(PopupFlyoutBase.OverlayInputPassThroughElementProperty, value, ValueNone)
+        
+        static member onPopupHostChanged<'t when 't :> PopupFlyoutBase>(func: IPopupHost voption -> unit, ?subPatchOptions) : IAttr<'t> =
+            let name = (nameof Unchecked.defaultof<IPopupHostProvider>.add_PopupHostChanged).Substring(4)
+            let factory: AvaloniaObject * (IPopupHost voption -> unit) * CancellationToken -> unit = 
+                (fun (control, func, token) ->
+                    let control = (control :?> 't) :> IPopupHostProvider
+                    let hander (host:IPopupHost) =
+                        match host with
+                        | null -> func ValueNone
+                        | host -> func (ValueSome host)
+
+                    control.add_PopupHostChanged hander
+                    token.Register(fun () -> control.remove_PopupHostChanged hander) |> ignore)
+            
+            AttrBuilder<'t>.CreateSubscription(name, factory, func, ?subPatchOptions = subPatchOptions)
+
+        static member onOpening<'t when 't :> PopupFlyoutBase>(func: 't -> unit, ?subPatchOptions) : IAttr<'t> =
+            let name = nameof Unchecked.defaultof<'t>.Opening
+            let factory: AvaloniaObject * ('t -> unit) * CancellationToken -> unit = 
+                (fun (control, func, token) ->
+                    let control = control :?> 't
+                    let hander = EventHandler(fun s _ -> s :?> 't |> func)
+                    let event = control.Opening
+
+                    event.AddHandler(hander)
+                    token.Register(fun () -> event.RemoveHandler(hander)) |> ignore)
+            
+            AttrBuilder<'t>.CreateSubscription(name, factory, func, ?subPatchOptions = subPatchOptions)
+
+        static member onClosing<'t when 't :> PopupFlyoutBase>(func: 't * CancelEventArgs -> unit, ?subPatchOptions) : IAttr<'t> =
+            let name = nameof Unchecked.defaultof<'t>.Closing
+            let factory: AvaloniaObject * ('t * CancelEventArgs -> unit) * CancellationToken -> unit = 
+                (fun (control, func, token) ->
+                    let control = control :?> 't
+                    let hander = EventHandler<_>(fun s e -> func(s :?> 't, e))
+                    let event = control.Closing
+
+                    event.AddHandler(hander)
+                    token.Register(fun () -> event.RemoveHandler(hander)) |> ignore)
+            
+            AttrBuilder<'t>.CreateSubscription(name, factory, func, ?subPatchOptions = subPatchOptions)
```

Comment:
Should I split it into FlyoutBase.fs/ PopupFlyoutBase.fs?

Reply:
We have MenuBase in its own file so that would be consistent with menus, but I don't know how important that part really is.